src/packages/documents/document-types/workspace/views/design/document-type-workspace-view-edit-properties.element.ts

```diff
@@ -77,48 +82,89 @@ export class UmbDocumentTypeWorkspaceViewEditPropertiesElement extends UmbLitEle
 	@state()
 	_propertyStructure: Array<PropertyTypeModelBaseModel> = [];
 
+	@state()
+	_ownerDocumentTypes?: DocumentTypeResponseModel[];
+
+	@state()
+	protected _modalRouteNewProperty?: string;
+
 	constructor() {
 		super();
 
 		this.consumeContext(UMB_WORKSPACE_CONTEXT, (workspaceContext) => {
 			this._propertyStructureHelper.setStructureManager(
-				(workspaceContext as UmbDocumentTypeWorkspaceContext).structure
+				(workspaceContext as UmbDocumentTypeWorkspaceContext).structure,
 			);
 		});
 		this.observe(this._propertyStructureHelper.propertyStructure, (propertyStructure) => {
 			this._propertyStructure = propertyStructure;
 			this.#propertySorter.setModel(this._propertyStructure);
 		});
+
+		// Note: Route for adding a new property
+		new UmbModalRouteRegistrationController(this, UMB_PROPERTY_SETTINGS_MODAL)
+			.addAdditionalPath('new-property')
+			.onSetup(() => {
+				return {};
+			})
+			.onSubmit((result) => {
+				this.#addProperty(result);
+			})
+			.observeRouteBuilder((routeBuilder) => {
+				this._modalRouteNewProperty = routeBuilder(null);
+			});
 	}
 
-	async #onAddProperty() {
-		const property = await this._propertyStructureHelper.addProperty(this._containerId);
-		if (!property) return;
+	connectedCallback(): void {
+		super.connectedCallback();
+		const doctypes = this._propertyStructureHelper.getOwnerDocumentTypes();
+		if (!doctypes) return;
+		this.observe(doctypes, (documents) => {
+			this._ownerDocumentTypes = documents;
+		});
```

Comment:
Since connectedCallback can happen multiple times, we should add an identifier to this observe. its a third argument like , 'observeOwnerDocumentTypes')